php/WP_CLI/Request.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace WP_CLI;
+
+/**
+ * Interprets CLI arguments to a structured representation
+ */
+class Request implements \ArrayAccess {
```

Comment:
Does it make sense to implement `ArrayAccess` as a utility feature, given this intends to solve the ambiguity problem in the first place?